homeassistant/components/rainbird/config_flow.py

```diff
@@ -120,18 +127,23 @@ async def _test_connection(self, host: str, password: str) -> str:
 
     async def async_finish(
         self,
-        serial_number: str,
         data: dict[str, Any],
         options: dict[str, Any],
     ) -> FlowResult:
         """Create the config entry."""
-        # Prevent devices with the same serial number. If the device does not have a serial number
-        # then we can at least prevent configuring the same host twice.
-        if serial_number:
-            await self.async_set_unique_id(serial_number)
-            self._abort_if_unique_id_configured()
-        else:
-            self._async_abort_entries_match(data)
+        # The integration has historically used a serial number, but not all devices
+        # historically had a valid one. Now the mac address is used as a unique id
+        # and serial is still persisted in config entry data in case it is needed
+        # in the future.
+        # Either way, also prevent configuring the same host twice.
+        await self.async_set_unique_id(format_mac(data[CONF_MAC]))
+        self._abort_if_unique_id_configured()
```

Comment:
I suggest we pass the `updates` parameter with host and password to allow the user to update that data for a existing device.

Reply:
Done